ecosystem/indexer-grpc/indexer-grpc-fullnode/src/convert.rs

```diff
@@ -835,6 +859,25 @@ pub fn convert_transaction(
         Transaction::ValidatorTransaction(api_validator_txn) => {
             convert_validator_transaction(api_validator_txn)
         },
+        Transaction::AutomatedTransaction(at) => {
+            timestamp = Some(convert_timestamp_usecs(at.timestamp.0));
+            let expiration_timestamp_secs = Some(convert_timestamp_secs(std::cmp::min(
+                at.meta.expiration_timestamp_secs.0,
+                chrono::NaiveDateTime::MAX.timestamp() as u64,
+            )));
```

Comment:
@aregng I am not sure I quite understand what is happening here. `min` of `MAX` time and `at.meta.expiration_timestamp_secs.0` would always result in `at.meta.expiration_timestamp_secs.0`, right? Not sure what is happening here.

Reply:
My assumption will be that if user specified timestamp exceeds expected MAX of the  `chrono::NaiveDateTime::MAX.timestamp()` it will take the later one. This part might require revision as AutoamtedTransactions will never be posted by users. I need to spend more time to understand APTOS-API and how it is utilized to make sure that these changes were valid or invalid.